File: app/src/main/java/com/chooloo/www/callmanager/ui/helpers/ListItemHolder.java
package com.chooloo.www.callmanager.ui.helpers;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.chooloo.www.callmanager.ui.listitem.ListItem;
import com.chooloo.www.callmanager.ui.listitem.ListItemPerson;

public class ListItemHolder<LI extends ListItem> extends RecyclerView.ViewHolder {

    private final LI mListItem;

    /**
     * Constructor
     *
     * @param listItem the layout view
     */
    public ListItemHolder(@NonNull LI listItem) {
        super(listItem);
        this.mListItem = listItem;
    }

    public LI getListItem() {
        return mListItem;
    }
}

File: chooloolib/src/main/java/com/chooloo/www/chooloolib/ui/widgets/listitem/ListItemHolder.kt
package com.chooloo.www.chooloolib.ui.widgets.listitem

import android.content.Context
import androidx.recyclerview.widget.RecyclerView

class ListItemHolder(val listItem: ListItem) : RecyclerView.ViewHolder(listItem) {
    constructor(context: Context) : this(ListItem(context))
}